mesonbuild/optinterpreter.py

```diff
@@ -77,9 +77,6 @@ def __init__(self, subproject, command_line_options):
             if self.subproject != '': # Strip the beginning.
                 if not o.startswith(self.sbprefix):
                     continue
-            else:
-                if ':' in o:
-                    continue
```

Comment:
Also, the failing test ensures that we won't have an option key (name) with a colon in it.